src/processor/operator/aggregate/hash_aggregate.cpp

```diff
@@ -117,6 +122,9 @@ void HashAggregate::executeInternal(ExecutionContext* context) {
     while (children[0]->getNextTuple(context)) {
         const auto numAppendedFlatTuples = localState.append(aggInputs, resultSet->multiplicity);
         metrics->numOutputTuple.increase(numAppendedFlatTuples);
+        if (sharedState->increaseAndCheckLimitCount(numAppendedFlatTuples)) {
```

Comment:
Better add a comment here to explain that the limit will only be applied if this is distinct hash aggregate. It's not to figure that out inside the operator without checking the limit push down optimizer.